jx install fails with SIGSEGV after creating default environments

### Summary 
Installing Jenkins X on a minikube kubernetes cluster with bitbucket server as git server. After helm chard deploys succesfully, segmentation violation encountered after printing `Creating default staging and production environments`.

### Steps to reproduce the behavior
1. Using minikube version 0.32.0
1. Create cluster with `minikube start --vm-driver kvm2 --cpus 8 --memory 24576 --disk-size 40g`
1. Deploy Bitbucket server to minikube cluser using manifests at https://github.com/wfhartford/bitbucket-kubernetes
1. Edit `/etc/hosts` file so that `jx.minikube.local` names are mapped to minikube host.
1. Install Jenkins X with command line:
```
jx install --provider=kubernetes --on-premise --git-provider-url=http://bitbucket-http.bitbucket --git-provider-kind=bitbucketserver --git-api-token=[redacted] --git-username=jenkins --domain=minikube.local
```
1. Accept default options for prompts
1. Wait for all pods to come up.

### Jx version

The output of `jx version` is:

```
NAME               VERSION
jx                 1.3.728
jenkins x platform 0.0.3193
Kubernetes cluster v1.12.4
kubectl            v1.13.1
helm client        v2.9.1+g20adb27
helm server        v2.9.1+g20adb27
git                git version 2.17.1
Operating System   Linux Mint 19.1 Tessa
```

### Kubernetes cluster

Minikube: `minikube start --vm-driver kvm2 --cpus 8 --memory 24576 --disk-size 40g`

### Operating system / Environment

Linux Mint 19.1

### Expected behavior
The `jx install` command completes successfully.

### Actual behavior
Output of `jx install`:
```
Namespace jx created 
 Using helmBinary helm with feature flag: none
Context "minikube" modified.
Storing the kubernetes provider kubernetes in the TeamSettings
Git configured for user: Wesley Hartford and email wesley@ze.com
Using helm2
Configuring tiller
Tiller Deployment is running in namespace kube-system
nginx ingress controller already enabled
Lets set up a Git user name and API token to be able to perform CI/CD

? Do you wish to use jenkins as the local Git user for http://bitbucket-http.bitbucket server: (Y/? Do you wish to use jenkins as the local Git user for http://bitbucket-http.bitbucket server: Yes
Select the CI/CD pipelines Git server and user
? Do you wish to use http://bitbucket-http.bitbucket as the pipelines Git server: Yes
? Do you wish to use jenkins as the pipelines Git user for http://bitbucket-http.bitbucket server:? Do you wish to use jenkins as the pipelines Git user for http://bitbucket-http.bitbucket server: Yes
Setting the pipelines Git server http://bitbucket-http.bitbucket and user name jenkins.
Saving the Git authentication configurationCurrent configuration dir: /home/wesley/.jx
options.Flags.CloudEnvRepository: https://github.com/jenkins-x/cloud-environments
options.Flags.LocalCloudEnvironment: false
Cloning the Jenkins X cloud environments repo to /home/wesley/.jx/cloud-environments
? A local Jenkins X cloud environments repository already exists, recreate with latest? Yes
Current configuration dir: /home/wesley/.jx
options.Flags.CloudEnvRepository: https://github.com/jenkins-x/cloud-environments
options.Flags.LocalCloudEnvironment: false
Cloning the Jenkins X cloud environments repo to /home/wesley/.jx/cloud-environments
Enumerating objects: 1289, done.
Total 1289 (delta 0), reused 0 (delta 0), pack-reused 1289
No default password set, generating a random one
Generated helm values /home/wesley/.jx/extraValues.yaml
Creating Secret jx-install-config in namespace jx
Installing Jenkins X platform helm chart from: /home/wesley/.jx/cloud-environments/env-kubernetes
? Select Jenkins installation type: Static Master Jenkins
Waiting for tiller pod to be ready, service account name is tiller, namespace is jx, tiller namespace is kube-system
Waiting for cluster role binding to be defined, named tiller-role-binding in namespace jx
 tiller cluster role defined: cluster-admin in namespace jx
tiller pod running
? Pick workload build pack:  Kubernetes Workloads: Automated CI+CD with GitOps Promotion
Setting the team build pack to kubernetes-workloads repo: https://github.com/jenkins-x-buildpacks/jenkins-x-kubernetes.git ref: master
Installing jx into namespace jx
Adding values file /home/wesley/.jx/cloud-environments/env-kubernetes/myvalues.yaml
Adding values file /home/wesley/.jx/adminSecrets.yaml
Adding values file /home/wesley/.jx/extraValues.yaml
Adding values file /home/wesley/.jx/cloud-environments/env-kubernetes/secrets.yaml
Upgrading Chart 'upgrade --namespace jx --install --timeout 6000 --version 0.0.3193 --values /home/wesley/.jx/cloud-environments/env-kubernetes/myvalues.yaml --values /home/wesley/.jx/adminSecrets.yaml --values /home/wesley/.jx/extraValues.yaml --values /home/wesley/.jx/cloud-environments/env-kubernetes/secrets.yaml jenkins-x jenkins-x/jenkins-x-platform'
waiting for install to be ready, if this is the first time then it will take a while to download images
Jenkins X deployments ready in namespace jx


	********************************************************

	     NOTE: Your admin password is: [redacted]

	********************************************************

	
Getting Jenkins API Token
Created user admin API Token for Jenkins server jenkins.jx.minikube.local at http://jenkins.jx.minikube.local
Updating Jenkins with new external URL details http://jenkins.jx.minikube.local
Creating default staging and production environments
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x11a3fa0]

goroutine 1 [running]:
github.com/gfleury/go-bitbucket-v1.NewBitbucketAPIResponse(0x0, 0xc000b23a00, 0x0, 0x2f78060)
	/home/jenkins/go/pkg/mod/github.com/gfleury/go-bitbucket-v1@v0.0.0-20181102191809-4910839b609e/api_response.go:319 +0x50
github.com/gfleury/go-bitbucket-v1.(*DefaultApiService).GetProjects(0xc0009a7928, 0xc0010614c8, 0x0, 0x0, 0x0)
	/home/jenkins/go/pkg/mod/github.com/gfleury/go-bitbucket-v1@v0.0.0-20181102191809-4910839b609e/default_api.go:5378 +0x6ae
github.com/jenkins-x/jx/pkg/gits.(*BitbucketServerProvider).ListOrganisations(0xc0010872b0, 0x0, 0x0, 0x0, 0x1, 0xc000a43090)
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/gits/bitbucket_server.go:148 +0x321
github.com/jenkins-x/jx/pkg/gits.GetOrganizations(0x7f55a9192f40, 0xc0010872b0, 0xc000546e70, 0x7, 0xc0010872b0, 0xc0010872b0, 0x0)
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/gits/provider.go:280 +0x6f
github.com/jenkins-x/jx/pkg/jx/cmd.(*InstallOptions).buildGitRepositoryOptionsForEnvironments(0xc0002b3900, 0x35, 0x2b8d759, 0x2)
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/install.go:1209 +0x2b2
github.com/jenkins-x/jx/pkg/jx/cmd.(*InstallOptions).createEnvironments(0xc0002b3900, 0x2b8d759, 0x2, 0x0, 0x0)
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/install.go:2028 +0x17b
github.com/jenkins-x/jx/pkg/jx/cmd.(*InstallOptions).Run(0xc0002b3900, 0x0, 0x0)
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/install.go:560 +0xd45
github.com/jenkins-x/jx/pkg/jx/cmd.NewCmdInstall.func1(0xc000331400, 0xc00057ea80, 0x0, 0x7)
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/install.go:254 +0x70
github.com/spf13/cobra.(*Command).execute(0xc000331400, 0xc00057e9a0, 0x7, 0x7, 0xc000331400, 0xc00057e9a0)
	/home/jenkins/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:766 +0x2cc
github.com/spf13/cobra.(*Command).ExecuteC(0xc00080c000, 0x0, 0x0, 0x0)
	/home/jenkins/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852 +0x2fd
github.com/spf13/cobra.(*Command).Execute(0xc00080c000, 0xc0004d34a0, 0x2f85ea0)
	/home/jenkins/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800 +0x2b
github.com/jenkins-x/jx/cmd/jx/app.Run(0xc0007a9f88, 0xc0000c2058)
	/home/jenkins/go/src/github.com/jenkins-x/jx/cmd/jx/app/jx.go:17 +0xed
main.main()
	/home/jenkins/go/src/github.com/jenkins-x/jx/cmd/jx/jx.go:10 +0x22
```

I just ran into the same problem:
**`jx version`:**

**Console transcript**
```
[root@gw01 ~]# jx install --git-provider-url=http://bitbucket.devtools.svc.c1.example.com --verbose=true --on-premise --git-provider-kind=bitbucketserver
Using helmBinary helm with feature flag: none
Context "kubernetes-admin@kubernetes" modified.
? Cloud Provider kubernetes
Git configured for user: cicd and email brian@example.com
Using helm2
Configuring tiller
Tiller Deployment is running in namespace kube-system
existing ingress controller found, no need to install a new one
Waiting for external loadbalancer to be created and update the nginx-ingress-controller service in kube-system namespace
Using external IP: 10.10.0.41
You can now configure a wildcard DNS pointing to the new Load Balancer address 10.10.0.41

If you do not have a custom domain setup yet, Ingress rules will be set for magic DNS nip.io.
Once you have a custom domain ready, you can update with the command jx upgrade ingress --cluster
If you don't have a wildcard DNS setup then setup a DNS (A) record and point it at: 10.10.0.41 then use the DNS domain in the next input...
? Domain box1.example.com
nginx ingress controller installed and configured
Lets set up a Git user name and API token to be able to perform CI/CD

? Do you wish to use cicd as the local Git user for bitbucket.devtools.svc.example.com server: Yes
Select the CI/CD pipelines Git server and user
? Do you wish to use bitbucket.devtools.svc.example.com as the pipelines Git server: Yes
? Do you wish to use cicd as the pipelines Git user for bitbucket.devtools.svc.example.com server: Yes
Setting the pipelines Git server http://bitbucket.devtools.svc.example.com and user name cicd.
Saving the Git authentication configurationCurrent configuration dir: /root/.jx
options.Flags.CloudEnvRepository: https://github.com/jenkins-x/cloud-environments
options.Flags.LocalCloudEnvironment: false
Cloning the Jenkins X cloud environments repo to /root/.jx/cloud-environments
? A local Jenkins X cloud environments repository already exists, recreate with latest? Yes
Current configuration dir: /root/.jx
options.Flags.CloudEnvRepository: https://github.com/jenkins-x/cloud-environments
options.Flags.LocalCloudEnvironment: false
Cloning the Jenkins X cloud environments repo to /root/.jx/cloud-environments
Enumerating objects: 8, done.
Counting objects: 100% (8/8), done.
Compressing objects: 100% (6/6), done.
Total 1373 (delta 2), reused 6 (delta 2), pack-reused 1365
? Select Jenkins installation type: Serverless Jenkins
No default password set, generating a random one
Generated helm values /root/.jx/extraValues.yaml
Creating Secret jx-install-config in namespace jx
Installing Jenkins X platform helm chart from: /root/.jx/cloud-environments/env-kubernetes

Installing knative into namespace jx
Updating Helm repository...
Helm repository update done.
Upgrading Chart 'upgrade --namespace jx --install --force --timeout 6000 --set build.auth.git.username=kubernetes-admin --set build.auth.git.password= hASHiSfORlUNCH --set  --set tillerNamespace= knative-build jenkins-x/knative-build'

retrying after error:failed to run 'helm upgrade --namespace jx --install --force --timeout 6000 --set build.auth.git.username=kubernetes-admin --set build.auth.git.password= hASHiSfORlUNCH *****  --set tillerNamespace= knative-build jenkins-x/knative-build' command in directory '', output: 'Release "knative-build" does not exist. Installing it now.
Error: customresourcedefinitions.apiextensions.k8s.io "builds.build.knative.dev" already exists'

Updating Helm repository...
Helm repository update done.
Upgrading Chart 'upgrade --namespace jx --install --force --timeout 6000 --set build.auth.git.username=kubernetes-admin --set build.auth.git.password=hASHiSfORlUNCH --set  --set tillerNamespace= knative-build jenkins-x/knative-build'
Waiting for tiller pod to be ready, service account name is tiller, namespace is jx, tiller namespace is kube-system
Waiting for cluster role binding to be defined, named tiller-role-binding in namespace jx
 tiller cluster role defined: cluster-admin in namespace jx
tiller pod running
? Pick workload build pack:  Kubernetes Workloads: Automated CI+CD with GitOps Promotion
Setting the team build pack to kubernetes-workloads repo: https://github.com/jenkins-x-buildpacks/jenkins-x-kubernetes.git ref: master
Installing jx into namespace jx
Installing jenkins-x-platform version: 0.0.3321
Adding values file /root/.jx/cloud-environments/env-kubernetes/myvalues.yaml
Adding values file /root/.jx/adminSecrets.yaml
Adding values file /root/.jx/extraValues.yaml
Adding values file /root/.jx/cloud-environments/env-kubernetes/secrets.yaml
Upgrading Chart 'upgrade --namespace jx --install --timeout 6000 --version 0.0.3321 --values /root/.jx/cloud-environments/env-kubernetes/myvalues.yaml --values /root/.jx/adminSecrets.yaml --values /root/.jx/extraValues.yaml --values /root/.jx/cloud-environments/env-kubernetes/secrets.yaml jenkins-x jenkins-x/jenkins-x-platform'
waiting for install to be ready, if this is the first time then it will take a while to download images
Jenkins X deployments ready in namespace jx
Configuring the TeamSettings for Prow


	********************************************************

	     NOTE: Your admin password is: hASHiSfORbREAKFAST

	********************************************************

	
Creating default staging and production environments
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x11fbad0]

goroutine 1 [running]:
github.com/gfleury/go-bitbucket-v1.NewBitbucketAPIResponse(0x0, 0xc000b2fa00, 0x0, 0x371e800)
	/home/jenkins/go/pkg/mod/github.com/gfleury/go-bitbucket-v1@v0.0.0-20181102191809-4910839b609e/api_response.go:319 +0x50
github.com/gfleury/go-bitbucket-v1.(*DefaultApiService).GetProjects(0xc0004a2c38, 0xc0011c9460, 0x0, 0x0, 0x0)
	/home/jenkins/go/pkg/mod/github.com/gfleury/go-bitbucket-v1@v0.0.0-20181102191809-4910839b609e/default_api.go:5378 +0x6ae
github.com/jenkins-x/jx/pkg/gits.(*BitbucketServerProvider).ListOrganisations(0xc0007d8410, 0x0, 0x0, 0x0, 0x1, 0xc00027f700)
	/workspace/pkg/gits/bitbucket_server.go:148 +0x321
github.com/jenkins-x/jx/pkg/gits.GetOrganizations(0x7f6803486458, 0xc0007d8410, 0xc0005477b0, 0x4, 0xc0007d8410, 0xc0007d8410, 0x0)
	/workspace/pkg/gits/provider.go:280 +0x6f
github.com/jenkins-x/jx/pkg/jx/cmd.(*InstallOptions).buildGitRepositoryOptionsForEnvironments(0xc00013ec00, 0x35, 0x30696a7, 0x2)
	/workspace/pkg/jx/cmd/install.go:1217 +0x2b2
github.com/jenkins-x/jx/pkg/jx/cmd.(*InstallOptions).createEnvironments(0xc00013ec00, 0x30696a7, 0x2, 0x0, 0x0)
	/workspace/pkg/jx/cmd/install.go:2054 +0x183
github.com/jenkins-x/jx/pkg/jx/cmd.(*InstallOptions).Run(0xc00013ec00, 0x0, 0x0)
	/workspace/pkg/jx/cmd/install.go:571 +0xeb2
github.com/jenkins-x/jx/pkg/jx/cmd.NewCmdInstall.func1(0xc000949400, 0xc000812800, 0x0, 0x4)
	/workspace/pkg/jx/cmd/install.go:257 +0x70
github.com/spf13/cobra.(*Command).execute(0xc000949400, 0xc0008127c0, 0x4, 0x4, 0xc000949400, 0xc0008127c0)
	/home/jenkins/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:766 +0x2cc
github.com/spf13/cobra.(*Command).ExecuteC(0xc000172c80, 0x0, 0x0, 0x0)
	/home/jenkins/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852 +0x2fd
github.com/spf13/cobra.(*Command).Execute(0xc000172c80, 0xc000280000, 0x372edc0)
	/home/jenkins/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800 +0x2b
github.com/jenkins-x/jx/cmd/jx/app.Run(0xc000a57f88, 0xc0000560b8)
	/workspace/cmd/jx/app/jx.go:17 +0xed
main.main()
	/workspace/cmd/jx/jx.go:11 +0x22
```

--- a/go.mod
+++ b/go.mod
@@ -45,7 +45,7 @@ require (
 	github.com/fatih/color v1.7.0
 	github.com/fatih/structs v1.1.0
 	github.com/flynn/go-shlex v0.0.0-20150515145356-3f9db97f8568 // indirect
-	github.com/gfleury/go-bitbucket-v1 v0.0.0-20190216152406-3a732135aa4d
+	github.com/gfleury/go-bitbucket-v1 v0.0.0-20200320173742-022f4bab9090
 	github.com/ghodss/yaml v1.0.0
 	github.com/gliderlabs/ssh v0.1.1 // indirect
 	github.com/go-ole/go-ole v1.2.1 // indirect

--- a/pkg/gits/bitbucket_server.go
+++ b/pkg/gits/bitbucket_server.go
@@ -555,7 +555,7 @@ func (b *BitbucketServerProvider) toPullRequest(bPR bitbucket.PullRequest) *GitP
 		URL:   bPR.Author.User.Links.Self[0].Href,
 		Login: bPR.Author.User.Slug,
 		Name:  bPR.Author.User.Name,
-		Email: bPR.Author.User.Email,
+		Email: bPR.Author.User.EmailAddress,
 	}
 	answer := &GitPullRequest{
 		URL:           bPR.Links.Self[0].Href,
@@ -633,13 +633,13 @@ func convertBitBucketCommitToGitCommit(bCommit *bitbucket.Commit, repo *GitRepos
 		Author: &GitUser{
 			Login: bCommit.Author.Name,
 			Name:  bCommit.Author.DisplayName,
-			Email: bCommit.Author.Email,
+			Email: bCommit.Author.EmailAddress,
 		},
 		URL: repo.URL + "/commits/" + bCommit.ID,
 		Committer: &GitUser{
 			Login: bCommit.Committer.Name,
 			Name:  bCommit.Committer.DisplayName,
-			Email: bCommit.Committer.Email,
+			Email: bCommit.Committer.EmailAddress,
 		},
 	}
 }
@@ -1080,7 +1080,7 @@ func (b *BitbucketServerProvider) UserInfo(username string) *GitUser {
 	return &GitUser{
 		Login: username,
 		Name:  user.DisplayName,
-		Email: user.Email,
+		Email: user.EmailAddress,
 		URL:   user.Links.Self[0].Href,
 	}
 }

--- a/pkg/gits/bitbucket_server_test.go
+++ b/pkg/gits/bitbucket_server_test.go
@@ -125,6 +125,35 @@ func (suite *BitbucketServerProviderTestSuite) TestListOrganizations() {
 	suite.Require().NotEmpty(orgs)
 }
 
+// https://github.com/jenkins-x/jx/issues/2682
+// Issue #2682
+func (suite *BitbucketServerProviderTestSuite) TestListOrganizationsFailing() {
+	as := auth.AuthServer{}
+	ua := auth.UserAuth{}
+
+	git := gits.NewGitCLI()
+	bp, err := gits.NewBitbucketServerProvider(&as, &ua, git)
+
+	suite.Require().NotNil(bp)
+	suite.Require().Nil(err)
+
+	var ok bool
+	provider, ok := bp.(*gits.BitbucketServerProvider)
+	suite.Require().True(ok)
+	suite.Require().NotNil(suite.provider)
+
+	// Broken URL
+	cfg := bitbucket.NewConfiguration("http://127.0.0.1:4444/rest")
+	ctx := context.Background()
+
+	apiKeyAuthContext := context.WithValue(ctx, bitbucket.ContextAccessToken, ua.ApiToken)
+	provider.Client = bitbucket.NewAPIClient(apiKeyAuthContext, cfg)
+
+	orgs, err := provider.ListOrganisations()
+	suite.Require().EqualError(err, "Get http://127.0.0.1:4444/rest/api/1.0/projects?limit=25&start=0: dial tcp 127.0.0.1:4444: connect: connection refused")
+	suite.Require().Empty(orgs)
+
+}
 func (suite *BitbucketServerProviderTestSuite) TestListRepositories() {
 	repos, err := suite.provider.ListRepositories("TEST-ORG")
 
@@ -341,7 +370,7 @@ func (suite *BitbucketServerProviderTestSuite) TestUserInfo() {
 	suite.Require().Equal(gits.GitUser{
 		Login: "test-user",
 		Name:  "Test User",
-		Email: "",
+		Email: "test-user@example.com",
 		URL:   "http://auth.example.com/users/test-user",
 	}, *userInfo)
 }

